client/src/components/Header/index.tsx

```diff
@@ -15,6 +15,7 @@ interface IHeaderProps {
   onMouseEnter: () => void;
   onMouseLeave: () => void;
   imgSrc?: string;
+  onLogoutButtonClick: () => void;
```

Comment:
```suggestion
  onLogout: () => void;
```
외부에서는 `logout()` 함수만 넘겨주고 그것이 버튼을 클릭해서 발생하는 것인지 keypress 때문인지는 몰라도 될 것 같아요.
함수를 props로 할 때, 이벤트째로 넘겨주기보단 이벤트를 가공해서 넘겨주는 건 어떠신가요? ☺️

Reply:
깔끔하고 괜찮은거같네요! :)